exporter/local/export.go

```diff
@@ -117,20 +131,44 @@ func (e *localExporterInstance) Export(ctx context.Context, inp *exporter.Source
 				defer cleanup()
 			}
 
+			if !e.split {
+				err = outputFS.Walk(ctx, func(p string, fi os.FileInfo, err error) error {
+					if fi.IsDir() {
+						return nil
+					}
+					if err != nil && !errors.Is(err, os.ErrNotExist) {
+						return err
+					}
+					st, ok := fi.Sys().(*fstypes.Stat)
+					if !ok {
+						return errors.WithStack(&os.PathError{Path: p, Err: syscall.EBADMSG, Op: "fileinfo without stat info"})
+					}
+					if _, ok := visitedPath[st.Path]; ok {
+						return errors.Errorf("cannot overwrite %s with output split option disabled", p)
```

Comment:
If we could provide some more context here to users debugging failures here, that would be useful.

Something like `"path/to/file from linux/amd64 would be overwritten by linux/arm64"` - we can get the platform info by saving the `k string` from the `export` parameter and then saving that into `visitedPath`.